src/app/core/services/scheduling.service.ts

```diff
@@ -163,16 +168,28 @@ export class SchedulingService {
   }
 
   generateSchedule(force: boolean) {
+    const completedTasks = this.getCompletedTasks()
     const scheduleVProm = this.storage.get(StorageKeys.SCHEDULE_VERSION)
     const configVProm = this.storage.get(StorageKeys.CONFIG_VERSION)
     const refDate = this.storage.get(StorageKeys.REFERENCEDATE)
-
-    return Promise.all([scheduleVProm, configVProm, refDate]).then(data => {
-      this.scheduleVersion = data[0]
-      this.configVersion = data[1]
-      this.refTimestamp = data[2]
-      if (data[0] !== data[1] || force) {
+    const utcOffsetPrev = this.storage.get(StorageKeys.UTC_OFFSET_PREV)
+
+    return Promise.all([
+      completedTasks,
+      scheduleVProm,
+      configVProm,
+      refDate,
+      utcOffsetPrev
+    ]).then(data => {
+      this.completedTasks = data[0] ? data[0] : []
+      this.scheduleVersion = data[1]
+      this.configVersion = data[2]
+      this.refTimestamp = data[3]
+      this.utcOffsetPrev = data[4]
+      if (data[1] !== data[2] || force) {
         console.log('Changed protocol version detected. Updating schedule..')
```

Comment:
Better change this console message to just `updating schedule`.

Reply:
Made the changes.